3p/facebook.js

```diff
@@ -36,31 +36,65 @@ function getFacebookSdk(global, cb) {
 
 /**
  * @param {!Window} global
- * @param {!Object} data
+ * @param {!Object} data The element data
+ * @return {!Element} div
  */
-export function facebook(global, data) {
+function getPostContainer(global, data) {
+  const container = global.document.createElement('div');
   const embedAs = data.embedAs || 'post';
   user().assert(['post', 'video'].indexOf(embedAs) !== -1,
       'Attribute data-embed-as  for <amp-facebook> value is wrong, should be' +
       ' "post" or "video" was: %s', embedAs);
-  const fbPost = global.document.createElement('div');
-  fbPost.className = 'fb-' + embedAs;
-  fbPost.setAttribute('data-href', data.href);
-  global.document.getElementById('c').appendChild(fbPost);
+  container.className = 'fb-' + embedAs;
+  container.setAttribute('data-href', data.href);
+
+  return container;
+}
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data The element data
+ * @return {!Element} div
+ */
+function getCommentsContainer(global, data) {
+  const container = global.document.createElement('div');
+  container.className = 'fb-comments';
+  container.setAttribute('data-href', data.href);
+  container.setAttribute('data-numposts', data.numposts || 10);
+  container.setAttribute('data-order-by', data.orderBy || 'social');
+  container.setAttribute('data-width', '100%');
+
+  return container;
+}
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function facebook(global, data) {
+  const extension = global.context.tagName;
+  let container;
+
+  if (extension === 'AMP-FACEBOOK-COMMENTS') {
+    container = getCommentsContainer(global, data);
+  } else if (extension === 'AMP-FACEBOOK') {
+    container = getPostContainer(global, data);
+  }
+
+  global.document.getElementById('c').appendChild(container);
```

Comment:
Compiler won't allow this since `container` could be undefined based on the ifs above. You can assert the value, e.g. `user().assert()` or just throw out of final `else` for the `if` above.